// pixi-gl-core/src/shader/index.js
/*
 * react-twoo
 * undefined
 * https://github.com/aarondupon/react-twoo
 * v0.1.0
 * undefined License
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var compileProgram = require('./compileProgram.js');
var mapType = require('./mapType.js');
var mapSize = require('./mapSize.js');
var extractAttributes = require('./extractAttributes.js');
var defaultValue = require('./defaultValue.js');
var extractUniforms = require('./extractUniforms.js');
var setPrecision = require('./setPrecision.js');
var generateUniformAccessObject = require('./generateUniformAccessObject.js');

var shader = {
  compileProgram: compileProgram['default'],
  defaultValue: defaultValue['default'],
  extractAttributes: extractAttributes['default'],
  extractUniforms: extractUniforms['default'],
  generateUniformAccessObject: generateUniformAccessObject['default'],
  setPrecision: setPrecision['default'],
  mapSize: mapSize['default'],
  mapType: mapType['default']
};

exports.__moduleExports = shader;
exports.default = shader;
