// pixi-gl-core/src/index.js
/*
 * react-twoo
 * undefined
 * https://github.com/aarondupon/react-twoo
 * v0.1.0
 * undefined License
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _commonjsHelpers = require('../../../_virtual/_commonjsHelpers.js');
var createContext = require('./createContext.js');
var setVertexAttribArrays = require('./setVertexAttribArrays.js');
var GLBuffer = require('./GLBuffer.js');
var GLTexture = require('./GLTexture.js');
var GLFramebuffer = require('./GLFramebuffer.js');
var GLShader = require('./GLShader.js');
var VertexArrayObject = require('./VertexArrayObject.js');
var index = require('./shader/index.js');

var src = _commonjsHelpers.createCommonjsModule(function (module) {
var gl = {
  createContext: createContext['default'],
  setVertexAttribArrays: setVertexAttribArrays['default'],
  GLBuffer: GLBuffer['default'],
  GLFramebuffer: GLFramebuffer['default'],
  GLShader: GLShader['default'],
  GLTexture: GLTexture['default'],
  VertexArrayObject: VertexArrayObject['default'],
  shader: index['default']
}; // Export for Node-compatible environments

if ( module.exports) {
  // Export the module
  module.exports = gl;
} // Add to the browser window pixi.gl


if (typeof window !== 'undefined') {
  // add the window object
  window.PIXI = window.PIXI || {};
  window.PIXI.glCore = gl;
}
});

exports.__moduleExports = src;
exports.default = src;
